test/functional/interface_ipc.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python3
+# Copyright (c) The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test the IPC (multiprocess) interface."""
+import asyncio
+from io import BytesIO
+from pathlib import Path
+import shutil
+from test_framework.messages import (CBlock, CTransaction, ser_uint256)
+from test_framework.test_framework import (BitcoinTestFramework, assert_equal)
+
+# Test may be skipped and not have capnp installed
+try:
+    import capnp  # type: ignore[import] # noqa: F401
+except ImportError:
+    pass
+
+
+class IPCInterfaceTest(BitcoinTestFramework):
+
+    def skip_test_if_missing_module(self):
+        self.skip_if_no_ipc()
+        self.skip_if_no_py_capnp()
+
+    def load_capnp_modules(self):
+        # If pycapnp was installed with bundled capnp, capnp/c++.capnp can be found here.
+        bundled_capnp_dir = Path(capnp.__path__[0]).parent
+        # Otherwise guess the location of capnp/c++.capnp based on the installation path of the
+        # capnp binary itself.
+        bin_capnp_dir = Path(shutil.which("capnp")).parent.parent / "include"
+        src_dir = Path(self.config['environment']['SRCDIR']) / "src"
+        mp_dir = src_dir / "ipc" / "libmultiprocess" / "include"
+        imports = [str(bundled_capnp_dir), str(bin_capnp_dir), str(src_dir), str(mp_dir)]
+        return {
+            "proxy": capnp.load(str(mp_dir / "mp" / "proxy.capnp"), imports=imports),
+            "init": capnp.load(str(src_dir / "ipc" / "capnp" / "init.capnp"), imports=imports),
+            "echo": capnp.load(str(src_dir / "ipc" / "capnp" / "echo.capnp"), imports=imports),
+            "mining": capnp.load(str(src_dir / "ipc" / "capnp" / "mining.capnp"), imports=imports),
+        }
+
+    def set_test_params(self):
+        self.num_nodes = 1
+
+    def setup_nodes(self):
+        self.extra_init = [{"ipcbind": True}]
+        super().setup_nodes()
+        # Use this function to also load the capnp modules (we cannot use set_test_params for this,
+        # as it is being called before knowing whether capnp is available).
+        self.capnp_modules = self.load_capnp_modules()
+
+    async def make_capnp_init_ctx(self):
```

Comment:
088dc2c486af0ad6803d919d8114ab29d3cd652f: eventually I'd like to move this and `load_capnp_modules` into the test framework so that it's easier to expand e.g. `mining_template_verification.py` to call both `getblocktemplate proposal` and `checkBlock()`.